hack/pre-commit.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+GOFILES=$(git diff --cached --name-only --diff-filter=ACM | grep --color=never -E '^.*\.go$')
```

Comment:
**Question:** could we re-use this to only lint changed files on the pre-commit hook?

Reply:
I tried that but golangci-lint only runs on packages, not on single files.

Reply:
Ah that's unfortunate, but not a big deal.